htlcswitch/switch.go

```diff
@@ -366,6 +371,20 @@ func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID,
 		return zeroPreimage, err
```

Comment:
Consider moving the control tower up to the `router` level already in this pr. To finalize the required switch changes for reliable payments.

Reply:
Done. Note that there still might be some switch changes left (most notably making `SendHTLC` async), but I figured this PR was big enough as is now.